TACC/Core-Portal
# task/TV3-162: Upgrade Django to 3.2

## Overview

Updates Django to the 3.2 release, and addresses related deprecations and breaking changes.

## Related

* [TV3-162](https://jira.tacc.utexas.edu/browse/TV3-162)

## Changes

- Updates the python client to 3.11
- Updates poetry to 1.4.3
- Uses the built-in `models.JSONField`, and creates squashed migrations for future removal of the custom `fields.py` files

## Testing

1. Run `make build`
2. Run migrations
3. Logout the back in to cep.test
4. Ensure everything works as expected. Monitor docker logs for uncaught exceptions
